oot.sh

```diff
@@ -0,0 +1,137 @@
+}
+
+
+while getopts ':c:f' OPT; do
+	case ${OPT} in
+	c)
+		case ${OPTARG} in
+		clang|gcc|linaro_gcc)
+			export _compiler=${OPTARG}
+			;;
+		*)
+			echo "Compiler '${OPTARG}' unknown or not implemented"
+			exit 1
+			;;
+		esac
+		;;
+	f)
+		export _fastboot_flash=true
+		;;
+	\?)
+		echo "Invalid option: -${OPTARG}"
+		usage
+	esac
+done
+
+shift $(( OPTIND - 1 ))
+
+if (( "$#" == 0 )); then
+	usage
+fi
+
+# Default to clang
+if [[ -z "${_compiler}" ]]; then
+	export _compiler=clang
+fi
+
+
+_self_dir=$(realpath "$(dirname "$0")")
+
+
+# Platform common
+export BOARD_KERNEL_BASE=0x00000000
+export BOARD_KERNEL_PAGESIZE=4096
+export BOARD_KERNEL_TAGS_OFFSET=0x01E00000
+export BOARD_RAMDISK_OFFSET=0x02000000
+
+for _device in "$@"; do
+	unset BOARD_KERNEL_CMDLINE _platform _has_dtbo _recovery_ramdisk
+
+	BOARD_KERNEL_CMDLINE+=" lpm_levels.sleep_disabled=1"
+	BOARD_KERNEL_CMDLINE+=" service_locator.enable=1"
+	BOARD_KERNEL_CMDLINE+=" androidboot.hardware=${_device}"
+
+	# Device specific
+	case ${_device} in
+	kirin|mermaid)
+		export _platform=ganges
+		;;
+	discovery|pioneer|voyager)
+		export _platform=nile
+		;;
+	bahamut|griffin)
+		export _platform=kumano
+		;;
+	pdx201)
+		export _platform=seine
+		;;
+	akari|akatsuki|apollo)
+		export _platform=tama
+		;;
+	lilac|maple|poplar)
+		export _platform=yoshino
+		;;
+	*)
+		echo "Device '${_device}' unknown or not implemented"
+		exit 1
+		;;
+	esac
+
+	# Platform specific
+	case ${_platform} in
+	nile|ganges)
+		_verity_file=build/target/product/security/verity.x509.pem
+		_verity_key_id=$(openssl x509 -in $_verity_file -text | grep keyid | sed 's/://g' | tr -d '[:space:]' | tr '[:upper:]' '[:lower:]' | sed 's/keyid//g')
+
+		BOARD_KERNEL_CMDLINE+=" veritykeyid=id:$_verity_key_id"
+		#BOARD_KERNEL_CMDLINE+=" earlycon=msm_serial_dm,0xc170000"
+		;;
+	kumano)
+		export _has_dtbo=true
+
+		BOARD_KERNEL_CMDLINE+=" msm_drm.blhack_dsi_display0=dsi_panel_somc_${_platform}_cmd:config0"
+		BOARD_KERNEL_CMDLINE+=" androidboot.bootdevice=1d84000.ufshc"
+		BOARD_KERNEL_CMDLINE+=" swiotlb=2048"
+		#BOARD_KERNEL_CMDLINE+=" earlycon=msm_geni_serial,0xa90000"
+		;;
+	seine)
+		export _has_dtbo=true
+		export _recovery_ramdisk=false
+
+		BOARD_KERNEL_CMDLINE+=" msm_drm.blhack_dsi_display0=dsi_panel_somc_${_platform}_cmd:config0"
+		#BOARD_KERNEL_CMDLINE+=" earlycon=msm_geni_serial,0x4a90000"
+		;;
+	tama)
+		export _has_dtbo=true
+
+		BOARD_KERNEL_CMDLINE+=" msm_drm.dsi_display0=dsi_panel_somc_${_platform}_cmd:config0"
+		BOARD_KERNEL_CMDLINE+=" androidboot.bootdevice=1d84000.ufshc"
+		BOARD_KERNEL_CMDLINE+=" swiotlb=2048"
+		#BOARD_KERNEL_CMDLINE+=" earlycon=msm_geni_serial,0xa84000 restore_msm_uart=0x03404000"
+		;;
+	yoshino)
+		_verity_file=build/target/product/security/verity.x509.pem
+		_verity_key_id=$(openssl x509 -in $_verity_file -text | grep keyid | sed 's/://g' | tr -d '[:space:]' | tr '[:upper:]' '[:lower:]' | sed 's/keyid//g')
+
+		BOARD_KERNEL_CMDLINE+=" veritykeyid=id:$_verity_key_id"
```

Comment:
We should merge this complicated duplication with `nile|ganges`, perhaps using an extra `case` statement above this one just for verity.

Reply:
I would've made two case statements independent of each other:

```bash
	# Verity
	case ${_platform} in
	nile|ganges|yoshino)
		# Verity stuff
	;;
	esac

	# Platform specific
	case ${_platform} in
		# The rest of `BOARD_KERNEL_CMDLINE` per identical platform
	esac
```

I don't really care about keeping commented out uart params, they are already in the Sony repo and never needed besides very early debug.

Reply:
Done as suggested :)